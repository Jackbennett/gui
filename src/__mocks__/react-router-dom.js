import React from 'react';
import { compose, setDisplayName } from 'recompose';
import * as rrd from 'react-router-dom';
// Just render plain div with its children
// eslint-disable-next-line no-import-assign
rrd.BrowserRouter = compose(setDisplayName('BrowserRouter'))(({ children }) => <div>{children}</div>);

module.exports = rrd;
