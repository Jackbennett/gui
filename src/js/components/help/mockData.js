export const helpProps = {
  docsVersion: '',
  findLocalIpAddress: jest.fn,
  isHosted: false,
  isEnterprise: false,
  menderDebPackageVersion: 'master',
  menderVersion: 'master',
  menderArtifactVersion: 'master',
  org: {
    tenant_token: ''
  }
};
