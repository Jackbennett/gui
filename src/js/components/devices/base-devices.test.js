import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import {
  AcceptedEmptyState,
  DeviceCreationTime,
  DeviceExpansion,
  DeviceStatusHeading,
  PendingEmptyState,
  PreauthorizedEmptyState,
  RejectedEmptyState,
  RelativeDeviceTime
} from './base-devices';
import { defaultState, undefineds } from '../../../../tests/mockData';

describe('smaller components', () => {
  [
    AcceptedEmptyState,
    DeviceCreationTime,
    DeviceExpansion,
    DeviceStatusHeading,
    PendingEmptyState,
    PreauthorizedEmptyState,
    RejectedEmptyState,
    RelativeDeviceTime
  ].forEach(Component => {
    it(`renders ${Component.displayName || Component.name} correctly`, () => {
      const { baseElement } = render(
        <MemoryRouter>
          <Component filters={[]} highlightHelp={true} limitMaxed={true} onClick={jest.fn} allCount={10} device={defaultState.devices.byId.a1} />
        </MemoryRouter>
      );
      const view = baseElement.firstChild.firstChild;
      expect(view).toMatchSnapshot();
      expect(view).toEqual(expect.not.stringMatching(undefineds));
    });
  });
});
