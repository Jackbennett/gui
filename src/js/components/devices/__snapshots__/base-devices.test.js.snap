// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smaller components renders AcceptedEmptyState correctly 1`] = `
<div
  class="dashboard-placeholder"
>
  <p>
    No devices found
  </p>
</div>
`;

exports[`smaller components renders DeviceCreationTime correctly 1`] = `-`;

exports[`smaller components renders DeviceExpansion correctly 1`] = `
<div
  class="bold flexbox center-aligned link-color margin-right-small uppercased"
  style="white-space: nowrap;"
>
  view details 
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M16.01 11H4v2h12.01v3L20 12l-3.99-4z"
    />
  </svg>
</div>
`;

exports[`smaller components renders DeviceStatusHeading correctly 1`] = `<div />`;

exports[`smaller components renders PendingEmptyState correctly 1`] = `
<div
  class="dashboard-placeholder"
>
  <p>
    There are no devices pending authorization
  </p>
  <p>
    Visit the 
    <a
      href="/help/get-started"
    >
      Help section
    </a>
     to learn how to connect devices to the Mender server.
  </p>
</div>
`;

exports[`smaller components renders PreauthorizedEmptyState correctly 1`] = `
<div
  class="dashboard-placeholder"
>
  <p>
    There are no preauthorized devices.
  </p>
  <p>
    Preauthorize devices
     so that when they come online, they will connect to the server immediately
  </p>
  <img
    alt="preauthorize"
    src="preauthorize.png"
  />
</div>
`;

exports[`smaller components renders RejectedEmptyState correctly 1`] = `
<div
  class="dashboard-placeholder"
>
  <p>
    There are no rejected devices
  </p>
</div>
`;

exports[`smaller components renders RelativeDeviceTime correctly 1`] = `
<div
  class=""
  title=""
>
  -
</div>
`;
